src/ompl/geometric/planners/bitstar/BITstar.h

```diff
@@ -0,0 +1,614 @@
+            bool isCostWorseThan(const ompl::base::Cost& a, const ompl::base::Cost& b) const;
+
+            /** \brief Compare whether cost a and cost b are equivalent by checking that neither a or b is better than the other. */
+            bool isCostEquivalentTo(const ompl::base::Cost& a, const ompl::base::Cost& b) const;
+
+            /** \brief Compare whether cost a and cost b are not equivalent by checking if either a or b is better than the other. */
+            bool isCostNotEquivalentTo(const ompl::base::Cost& a, const ompl::base::Cost& b) const;
+
+            /** \brief Compare whether cost a is better or equivalent to cost b by checking that b is not better than a. */
+            bool isCostBetterThanOrEquivalentTo(const ompl::base::Cost& a, const ompl::base::Cost& b) const;
+
+            /** \brief Compare whether cost a is worse or equivalent to cost b by checking that a is not better than b. */
+            bool isCostWorseThanOrEquivalentTo(const ompl::base::Cost& a, const ompl::base::Cost& b) const;
+
+            /** \brief Returns whether the cost is finite or not. By default calls std::isfinite on Cost::value(). */
+            bool isFinite(const ompl::base::Cost& cost) const;
+
+            /** \brief The better of two costs.*/
+            ompl::base::Cost betterCost(const ompl::base::Cost& a, const ompl::base::Cost& b) const;
+
+            /** \brief Combine 3 costs */
+            ompl::base::Cost combineCosts(const ompl::base::Cost& a, const ompl::base::Cost& b, const ompl::base::Cost& c) const;
+
+            /** \brief Combine 4 costs */
+            ompl::base::Cost combineCosts(const ompl::base::Cost& a, const ompl::base::Cost& b, const ompl::base::Cost& c, const ompl::base::Cost& d) const;
+
+            /** \brief Calculate the fractional change of cost "newCost" from "oldCost" relative to "oldCost", i.e., (newCost - oldCost)/oldCost. */
+            double fractionalChange(const ompl::base::Cost& newCost, const ompl::base::Cost& oldCost) const;
+
+            /** \brief Calculate the fractional change of cost "newCost" from "oldCost" relative to "refCost", i.e., (newCost - oldCost)/refCost. */
+            double fractionalChange(const ompl::base::Cost& newCost, const ompl::base::Cost& oldCost, const ompl::base::Cost& refCost) const;
+            ///////////////////////////////////////////////////////////////////
+
+            ///////////////////////////////////////////////////////////////////
+            //Helper functions to calculate parameters:
+            /** \brief Initialize the nearest-neighbour terms */
+            void initializeNearestTerms();
+
+            /** \brief Update the appropriate nearest-neighbour terms, r_ and k_ */
+            virtual void updateNearestTerms();
+
+            /** \brief Calculate the r for r-disc nearest neighbours, a function of the current graph */
+            double r(unsigned int N) const;
```

Comment:
Please use longer function names for this one and the one below. Even if it is something like rConstant